app/controllers/api/v0/service_offering_icons_controller.rb

```diff
@@ -4,6 +4,15 @@ class ServiceOfferingIconsController < ApplicationController
       include Api::V0::Mixins::IndexMixin
       include Api::V0::Mixins::ShowMixin
 
+      def icon_data
+        service_offering_icon = model.find(params[:service_offering_icon_id].to_i)
+
+        send_data(service_offering_icon.data, 
+          :type         => MimeMagic.by_magic(service_offering_icon.data).type,
+          :disposition  => 'inline')
+      rescue ActiveRecord::RecordNotFound
+        head :not_found
```

Comment:
Can we just let this blow all the way out instead of rescuing?

Reply:
This is interesting actually, not sure if it is intentional or not but all of the current index methods do this for the primary collect, but the subcollection uses the `rescue_from` error document from the base ApplicationController.

Example index spec for the primary collection: https://github.com/ManageIQ/topological_inventory-api/blob/master/spec/requests/api/v0.1/shared_examples_for_index.rb#L45-L54
Example index spec for the subcollection: https://github.com/ManageIQ/topological_inventory-api/blob/master/spec/requests/api/v0.1/shared_examples_for_index.rb#L88-L99

@abellotti do you know if this is intentional?  It would seem that if we remove all of these it would be more consistent.

Reply:
At least we should start to be consistent with the swagger file, which says we should return 404, not 400.

Reply:
hm, ok we return 400 only in ServicePlanController, the rest is ok

Reply:
not sure of the intent/historical reason, but from what I can tell, if we nix the rescue's and just leverage the rescue_from in the app controller for the ActiveRecord::RecordNotFound errors, instead of getting an empty response body with the 404, you'd get the:

```
{
  "errors" : [
     { "details" => "Couldn't find <type> with 'id'=....,
        "status" => 404
      }
  ]
}
```

Which isn't entirely bad. The specs would just need updating for the different expectations, but like you said, more consistent.

Reply:
:+1: @abellotti yeah I agree, I'll work on updating the specs

Reply:
@abellotti https://github.com/ManageIQ/topological_inventory-api/pull/164